### PR TITLE
[Relay][Training] Add gradient for cast

### PR DESCRIPTION
**Pull request by MarisaKirisame**
@SWu @vinx13 @yuluny2 can you guys help review?

### CODE REVIEW

File: python/tvm/relay/op/_tensor_grad.py [hunk 1]
```diff
@@ -283,6 +284,13 @@ def dense_grad(orig, grad):
             collapse_sum_like(data * transpose(grad), weight)]
 
 
+@register_gradient("cast")
+def cast_grad(orig, grad):
+    x = orig.args[0]
+    grad = cast_like(grad, x)
+    return [broadcast_to_like(grad, x)]
```
Comment (vinx13) [inline python/tvm/relay/op/_tensor_grad.py]: why do you need `broadcast_to_like`?

Comment (MarisaKirisame) [reply python/tvm/relay/op/_tensor_grad.py]: fixed

**Comment by MarisaKirisame** [comment]
@vinx13 @tqchen can you guys give another round of review? this pr is blocking another gradient pr, so I want this merged ASAP.

**Comment by vinx13** [comment]
Thanks @MarisaKirisame @SWu, this has been merged

**Pull request merged**